src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -230,6 +244,9 @@ ComponentRecipe getPackageRecipe(@NonNull ComponentIdentifier pkgId) throws Pack
     void deleteComponent(@NonNull ComponentIdentifier compId) throws PackageLoadingException {
         logger.atDebug("delete-component-start").kv("componentIdentifier", compId).log();
         IOException exception = null;
+        // issues #1111
+        // delete docker image before removing the recipe file
+        removeDockerImage(compId);
```

Comment:
I think we can do this in a much better way which is properly extensible.

I'd want to add a new method to the `ArtifactDownloader` interface which is for cleaning up artifacts. Then, here you will use `getArtifactDownloader` to get the proper downloader for each artifact of the component, then call the cleanup method.